src/batch.jl

```diff
@@ -153,11 +153,10 @@ end
         $launch_quote
         start = stop
       end
-      Nr -= nthread
+      Nr = (Nr - nthread) % Int
```

Comment:
This shouldn't be changing the type of `Nr`.
Please make sure `Nr` is initially an `Int`.